pkg/provider/gitlab/gitlab.go

```diff
@@ -564,23 +564,38 @@ func (v *Provider) fetchChangedFiles(_ context.Context, runevent *info.Event) (c
 			}
 		}
 	case triggertype.Push:
-		pushChanges, _, err := v.Client().Commits.GetCommitDiff(v.sourceProjectID, runevent.SHA, &gitlab.GetCommitDiffOptions{})
-		if err != nil {
-			return changedfiles.ChangedFiles{}, err
-		}
-		for _, change := range pushChanges {
-			changedFiles.All = append(changedFiles.All, change.NewPath)
-			if change.NewFile {
-				changedFiles.Added = append(changedFiles.Added, change.NewPath)
+		options := gitlab.GetCommitDiffOptions{}
+		pageOpts := []gitlab.RequestOptionFunc{}
```

Comment:
is there an override somewhere for the pagesize? The default pagesize of GL is 20 (eg https://docs.gitlab.com/api/rest/#keyset-based-pagination), and esp when the goal is to get all pages anyway, the page size should most likely be set to the maximum of 100 to reduce the number of api queries; this also seems like something that should be solved once all over the pac codebase 🤔